### PR TITLE
bindinfo: support drop bindings of dropped table

### PR DESCRIPTION
**Pull request by lamxTyler**
### What problem does this PR solve? 

When a table dropped, we cannot drop the bindings related to that table.

### What is changed and how it works?

We cannot drop the bindings because previously we need to genenate hints from it. Now this pr only compares the bind sql when drop a specific bindings.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - None

Related changes

 - None

Release note

 - None

### CODE REVIEW

File: bindinfo/handle.go [hunk 1]
```diff
@@ -522,19 +512,16 @@ func (h *BindHandle) insertBindInfoSQL(orignalSQL string, db string, info Bindin
 	)
 }
 
-func (h *BindHandle) logicalDeleteBindInfoSQL(originalSQL, db string, updateTs types.Time, bindingSQLs []string) string {
+func (h *BindHandle) logicalDeleteBindInfoSQL(originalSQL, db string, updateTs types.Time, bindingSQLs string) string {
```
Comment (lzmhhh123) [inline bindinfo/handle.go]: %s/bindingSQLs/bindingSQL

**Comment by winoros** [comment]
:joy: Seems that this pr is not related with `dropped table`？

**Comment by lamxTyler** [comment]
Why? Check the test case: https://github.com/pingcap/tidb/pull/14071/files#diff-8d7989adda266ebd5efd41c7814cc288R540

File: bindinfo/cache.go [hunk 2]
```diff
@@ -56,6 +56,14 @@ type Binding struct {
 	id string
 }
 
+func (b *Binding) isSame(rb *Binding) bool {
+	if b.id != "" && rb.id != "" {
+		return b.id == rb.id
+	}
+	// Sometimes we cannot construct `id` because of the changed schema, so we need to compare by bind sql.
+	return b.BindSQL == rb.BindSQL
```
Comment (winoros) [inline bindinfo/cache.go]: So can we directly compare the SQL instead of the id?

Comment (lamxTyler) [reply bindinfo/cache.go]: Users may write hints in different ways but acutally there are the same, so `id` is a way to handle the duplicates. And we we evolve bindings, it is important to not evolve existing bindings.

Comment (winoros) [reply bindinfo/cache.go]: I got you.

**Pull request merged**